File: estrutura_de_dados/python/stack.py
class Stack:
    def __init__(self):
        self.__data = []

    def push(self, item):
        self.__data.append(item)

    def pop(self):
        return self.__data.pop(-1)
    
    def top(self):
        return self.__data[-1]
    
    def empty(self):
        return not len(self.__data) > 0

File: estrutura_de_dados/typescript/stack.ts
class Stack {

    private data: any[];

    constructor(){}

    push(item: any): void {
        this.data.push(item);
    }

    pop(): any {
        return this.data.pop();
    }

    top(): any {
        return this.data[0];
    }

    empty(): boolean {
        return this.data.length === 0;
    }
}